src/Tools/dotnet-dsrouter/DiagnosticsServerRouterCommands.cs

```diff
@@ -411,6 +435,23 @@ private static NetServerRouterFactory.CreateInstanceDelegate ChooseTcpServerRout
             return tcpServerRouterFactory;
         }
 
+        private static void logDiagnosticPortsConfiguration(string deviceName, string deviceTcpIpAddress, bool deviceListenMode, string verbose)
+        {
+            StringBuilder message = new();
+
+            if (!string.IsNullOrEmpty(verbose))
+            {
+                deviceName = !string.IsNullOrEmpty(deviceName) ? $" on {deviceName} " : " ";
+                message.AppendLine($"Start an application{deviceName}with one of the following environment variables set:");
+            }
+
+            string listenMode = deviceListenMode ? ",listen" : "";
+            message.AppendLine($"DOTNET_DiagnosticPorts={deviceTcpIpAddress},nosuspend{listenMode}");
+            message.AppendLine($"DOTNET_DiagnosticPorts={deviceTcpIpAddress},suspend{listenMode}");
```

Comment:
Does the 'nosuspend' option make sense if `deviceListenMode=true`?

Reply:
Yes, the default in EventPipe is nosuspend, suspending at startup is normally considered a "special case" and only used when doing startup tracing. The default IPC listeners used on EventPipe desktop builds (Windows, Linux, MacOS, MacCatalyst) is running in listener mode using nosuspend.

Reply:
Second question: are all 4 variations ("suspend,listen", "suspend", "nosuspend,listen", "nosuspend") valid on all platforms? I was under the impression that "listen" was required on iOS device, although I don't remember the technical details.

Reply:
> Second question: are all 4 variations ("suspend,listen", "suspend", "nosuspend,listen", "nosuspend") valid on all platforms? I was under the impression that "listen" was required on iOS device, although I don't remember the technical details.

You can run all modes, but by default we used listen mode on iOS device since it works with usbmux, but you can set it up in connect mode as well, but you need to make sure the iOS device can connect to the tcp/ip address,port bound to dsrouter. We don't recommend it since you need to use a none loopback interface on the development machine, so that's why we choose listen mode + usbmux as default.

Reply:
@rolfbjarne Is there any variant when running app on iOS device where you can set it up to connect using 127.0.0.1:port and be automatically routed to a dev pc that it has been paired with (https://help.apple.com/xcode/mac/9.0/index.html?localePath=en.lproj#/devbc48d1bad)? If not, then I assume the only alternative when running on device using connect mode, is to use an IP address that can be routed to the development machine, run dsrouter bound to a IP address that can be accessed from iOS device wifi interface (and make sure the traffic finds it way through the firewall).

Reply:
> @rolfbjarne Is there any variant when running app on iOS device where you can set it up to connect using 127.0.0.1:port and be automatically routed to a dev pc that it has been paired with ([help.apple.com/xcode/mac/9.0/index.html?localePath=en.lproj#/devbc48d1bad](https://help.apple.com/xcode/mac/9.0/index.html?localePath=en.lproj#/devbc48d1bad))?

Not that I know of.

> If not, then I assume the only alternative when running on device using connect mode, is to use an IP address that can be routed to the development machine, run dsrouter bound to a IP address that can be accessed from iOS device wifi interface and make sure the traffic finds it way through the firewall.

Yes, that sounds like it could work, although not trivial. Note that iOS apps need a specific entitlement to access the local network (which is totally doable, it's just another hurdle).

Are there any features that require the connect mode?

Reply:
No, with dotnet-dsrouter we can handle all features using listen mode on device, so connect mode is not needed.

Reply:
I think it would make sense to only show the environment variable for the listen mode then for ios/tvos device, this is supposed to make things easier, not make people go into rabbit holes. If someone really wanted/needed the connect mode, I assume they could still use spell out all the arguments on the command line like they currently would have to.

Reply:
Right now it will display the mode used to use the running instance of dotnet-dsrouter, if you use ios-sim default it will use connect mode and only list that and if you use device mode it will use listen mode and only list that. If we only want to default to listen mode for both ios-sim and ios then we would need to make sure you can bind and connect to loopback interface on ios-simulator using for example 127.0.0.1:9000 and then connect to that from dotnet-dsrouter using 127.0.0.1:9000, I havn't tested that on ios-sim, but if that works as expected we could switch to listen mode for both simulator and device.

Reply:
If the default doesn't match your scenario, you can always spell out all the arguments needed like currently done, so no change there, the new defaults will just simplify the process to configure dotnet-dsrouter for the most common used scenarios.

Reply:
> if you use ios-sim, it will default to use connect mode and only list that and if you use device mode it will use listen mode and only list that

The code seems to print both, or am I missing something?

```cs
            message.AppendLine($"DOTNET_DiagnosticPorts={deviceTcpIpAddress},nosuspend{listenMode}");
            message.AppendLine($"DOTNET_DiagnosticPorts={deviceTcpIpAddress},suspend{listenMode}");
```

Reply:
Ah, I see, I'm getting confused between nosuspend/suspend and listen/nolisten.

Reply:
Yes, it will print two options, since you can run in nosuspend and suspend mode (used for startup tracing), so for example if you run with ios-sim it will print:

DOTNET_DiagnosticPorts=127.0.0.1,nosuspend,connect
DOTNET_DiagnosticPorts=127.0.0.1,suspend,connect

and with ios it will print:

DOTNET_DiagnosticPorts=127.0.0.1,nosuspend,listen
DOTNET_DiagnosticPorts=127.0.0.1,suspend,listen

the app can be launched with any version of the env variable for each scenario, but if you use the nosuspend mode the app won't wait for tools to connect during startup.